Sources/App/GithubService+IssueComment.swift

```diff
@@ -0,0 +1,89 @@
+import Vapor
+import Stevenson
+
+struct PingAction: Content {
+    let zen: String
+}
+
+struct CommentAction: Content {
+    let action: String
+    let comment: Comment
+    let issue: Issue
+
+    struct Comment: Content {
+        let body: String
+    }
+
+    struct Issue: Content {
+        let number: Int
+    }
+
+    struct PullRequest: Content {
+        let url: URL
+    }
+}
+
+extension GitHubService {
+    func webhook<T: Decodable>(from request: Request) throws -> Future<T> {
+        let headers = request.http.headers
+
+        guard
+            headers.firstValue(name: .userAgent)?.hasPrefix("GitHub-Hookshot/") == true,
+            headers.firstValue(name: .init("X-GitHub-Event")) == "issue_comment"
+        else {
+            throw Abort(.badRequest)
+        }
+
+        return try request.content.decode(T.self)
+    }
+}
+
+extension GitHubService {
+    // Handle incoming webhook for issue comment
+    func issueComment(
+        on request: Request,
+        ci: CircleCIService
+    ) throws -> Future<Response> {
+        return try webhook(from: request).flatMap { (action: CommentAction) in
+            guard action.comment.body.hasPrefix("@ios-bot-babylon") else {
+                throw Abort(.badRequest)
+            }
+
+            let repo = RepoMapping.ios.repository
+
+            return try self.pullRequest(
+                number: action.issue.number,
+                in: repo,
+                on: request
+            ).flatMap { pullRequest in
+                let branch = pullRequest.head.ref
+
+                let textComponents = action.comment.body.split(separator: " ")
+                guard textComponents.count >= 2 else {
+                    throw Abort(.badRequest)
+                }
```

Comment:
We should move this `guard` before the `self.pullRequest`. No need to fetch the PR metadata if we can know the comment is invalid beforehand.